client/blocks/login/index.jsx

```diff
@@ -402,42 +398,18 @@ class Login extends Component {
 			locale,
 		} = this.props;
 
-		if ( twoFactorEnabled && twoFactorAuthType === 'webauthn' && this.state.isBrowserSupported ) {
+		if ( twoFactorEnabled ) {
 			return (
-				<div>
-					<SecurityKeyForm twoFactorAuthType="webauthn" onSuccess={ this.handleValid2FACode } />
-				</div>
-			);
-		}
-
-		let poller;
-		if ( twoFactorEnabled && twoFactorAuthType && twoFactorNotificationSent === 'push' ) {
-			poller = <PushNotificationApprovalPoller onSuccess={ this.rebootAfterLogin } />;
-		}
-
-		if ( twoFactorEnabled && includes( [ 'authenticator', 'sms', 'backup' ], twoFactorAuthType ) ) {
-			return (
-				<div>
-					{ poller }
-					<VerificationCodeForm
-						isJetpack={ isJetpack }
-						isGutenboarding={ isGutenboarding }
-						onSuccess={ this.handleValid2FACode }
-						twoFactorAuthType={ twoFactorAuthType }
-					/>
-				</div>
-			);
-		}
-
-		if ( twoFactorEnabled && twoFactorAuthType === 'push' ) {
-			return (
-				<div>
-					{ poller }
-					<WaitingTwoFactorNotificationApproval
-						isJetpack={ isJetpack }
-						isGutenboarding={ isGutenboarding }
-					/>
-				</div>
+				<AsyncLoad
+					require="blocks/login/two-factor-authentication/two-factor-content"
+					isBrowserSupported={ this.isBrowserSupported }
```

Comment:
@sgomes looks like a typo on this line

`isBrowserSupported` is a property of `state`, not a class member. And looks like `webauthn` is the only auth type that uses that state, which is why it was the only one that failed.

If `<TwoFactorContent>` doesn't recognise any of the auth types then it fails to return anything, which causes that invariant failure: https://reactjs.org/docs/error-decoder.html/?invariant=152&args[]=B

Can't decide whether it's better to let it crash and get the error, or to add a `return null` branch to the `<TwoFactorContent>` component.

Reply:
Looks like you found the issue, thanks @p-jackson! 🙂 

I think it's best to `return null` there, instead of outputting an obscure error message to the console. Anyone in a future debugging scenario can still use React DevTools to see which component outputted `null`, and go from there.

Created a new PR with the fixes in #41223.